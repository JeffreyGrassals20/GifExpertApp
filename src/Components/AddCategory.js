import React,{useState} from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {


    const [inputValue, setInput] = useState('')

    const handleInputChange = (e) => {
        setInput(e.target.value)
    } 

    const handleSubmit = (e) =>{
        e.preventDefault() 

        if(inputValue.trim().length >= 1){
            setCategories(cats => [inputValue, ...cats])
            setInput('');
        }
    }
    return (
        <>
            <form onSubmit = {handleSubmit}>
            
                <h2>Add Category</h2>
                <input type="text" value = {inputValue} onChange = {handleInputChange}></input>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
