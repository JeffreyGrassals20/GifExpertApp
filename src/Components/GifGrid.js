import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import {GifGridItem} from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
   
    return (
        <>
        <h2>{category} </h2>
        {loading && <h5 className= 'card animate__animated animate__flash'>Loading...</h5>}

        { <div className = 'card-grid'>
                <h4>
                    {
                        images.map((img) => (
                         <GifGridItem  {...img} key={img.id}/>
                        ))
                    }
                </h4>

        </div> }

        </>
    )
}
