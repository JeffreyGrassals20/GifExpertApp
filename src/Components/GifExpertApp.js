import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp  =() => {

    const  brands  = ['Basketball'];

    const [categories, setCategories] = useState(brands)

    /*const handleAdd = () => {
        setCategories([...categories, 'new one'])
    }*/

    return (
        <>
            <h2> Gif Expert </h2>
            <AddCategory setCategories = {setCategories}/>
            <hr />
            <ol>
                {
                    categories.map( category => 
                        <GifGrid key={category} category={category}/>
                   )
                }
            </ol>
        </>
    )
    
}
export default GifExpertApp
