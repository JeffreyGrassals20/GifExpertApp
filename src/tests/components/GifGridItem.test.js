import React from 'react'
import {shallow} from 'enzyme'
import {GifGridItem} from '../../Components/GifGridItem'

describe('Gif Grid testing',() => {
    
    const title = 'The Title here'
    const url = 'https:/localhost/hello.png'
    const wrapper = shallow(<GifGridItem title={title} url = {url}/>) 

    test('Should Veirfy if component shows correctly',() => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should have a paragraph with the title ', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('should have image url', () => {
        const img = wrapper.find('img')
        expect(img.props().src).toBe(url)
    })

    test('should have an animate__Bounce', () => {
        const div = wrapper.find('div') 
        expect(div.prop('className')).toBe('card animate__animated animate__bounce')
    })
}) 