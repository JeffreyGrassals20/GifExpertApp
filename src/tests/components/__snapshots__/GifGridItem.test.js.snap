// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid testing Should Veirfy if component shows correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="The Title here"
    src="https:/localhost/hello.png"
  />
  <p>
    The Title here
  </p>
</div>
`;
